eth/handler_eth.go

```diff
@@ -43,6 +43,10 @@ func (h *ethHandler) TxPool() eth.TxPool { return h.txpool }
 
 // RunPeer is invoked when a peer joins on the `eth` protocol.
 func (h *ethHandler) RunPeer(peer *eth.Peer, hand eth.Handler) error {
+	// Cannot Handshake with a peer before finishing the bad hashes cleanup
+	if h.core.BadHashExistsInChain() {
+		return nil
```

Comment:
Same concern here, let the user know they can't handshake because they still have bad blocks.